mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php

```diff
@@ -218,26 +218,34 @@ public function getLatestTasks(
     ],
     $limit = Scheduler::TASK_BATCH_SIZE
   ) {
+    $result = [];
+    foreach ($statuses as $status) {
+      $tasksQuery = $this->doctrineRepository->createQueryBuilder('st')
+        ->select('st')
+        ->where('st.deletedAt IS NULL')
+        ->where('st.status = :status');
+
+      if ($status === ScheduledTaskEntity::VIRTUAL_STATUS_RUNNING) {
+        $tasksQuery = $tasksQuery->orWhere('st.status IS NULL');
+      }
 
-    $tasksQuery = $this->doctrineRepository->createQueryBuilder('st')
-      ->select('st')
-      ->where('st.deletedAt IS NULL')
-      ->where('st.status IN (:statuses)');
-
-    if (in_array(ScheduledTaskEntity::VIRTUAL_STATUS_RUNNING, $statuses)) {
-      $tasksQuery = $tasksQuery->orWhere('st.status IS NULL');
-    }
+      if ($type) {
+        $tasksQuery = $tasksQuery->andWhere('st.type = :type')
+          ->setParameter('type', $type);
+      }
 
-    if ($type) {
-      $tasksQuery = $tasksQuery->andWhere('st.type = :type')
-        ->setParameter('type', $type);
+      $tasks = $tasksQuery
+        ->setParameter('status', $status)
+        ->setMaxResults($limit)
+        ->orderBy('st.id', 'desc')
+        ->getQuery()
+        ->getResult();
+      bdump($tasks);
+      $result = array_merge($result, $tasks);
+      bdump($result);
```

Comment:
I don't think `bdump` should be here

Reply:
@triple0t Right. I forgot to push the latest changes! Thanks